extensions/wellinks/api/clients/wellinksGraphqlClient.ts

```diff
@@ -1,17 +1,18 @@
 import { GraphQLClient } from 'graphql-request'
 import { type settings } from '../../config/settings'
-
+import z from 'zod'
+const ClientSchema = z.object({
+  apiUrl: z.string(),
+  apiKey: z.string(),
+})
 export const initialiseClient = (
-  s: Record<keyof typeof settings, string | undefined>
+  s: Partial<Record<keyof typeof settings, string | undefined>>
 ): GraphQLClient | undefined => {
-  const { apiUrl, apiKey } = s
-  if (apiUrl !== undefined && apiKey !== undefined) {
-    return new GraphQLClient(apiUrl, {
-      headers: {
-        AuthorizationSource: 'API',
-        Authorization: `Basic ${apiKey}`,
-      },
-    })
-  }
-  return undefined
+  const { apiUrl, apiKey } = ClientSchema.parse(s)
```

Comment:
FYI when this parses, if the `apiUrl` and `apiKey` aren't there, it will throw a `ZodError`, meaning, this function will never return undefined. Instead, an error will propagate letting you know which fields are missing